biosys/apps/main/api/serializers.py

```diff
@@ -126,6 +126,16 @@ def __init__(self, instance=None, **kwargs):
         # from the species_naming_facade above.
         self.species_name_id_mapping_cached = None
 
+        # dynamic fields
+        request = ctx.get('request')
+        if request:
+            expected_fields = request.query_params.getlist('fields', [])
+            if expected_fields:
+                existing_fields = self.fields.keys()
+                excluded_fields = set(existing_fields) - set(expected_fields)
+                for field in excluded_fields:
+                    self.fields.pop(field)
```

Comment:
could probably do this in one line

```python
self.fields = {k: v for k: v in self.fields.items() if k in expected_fields]
```

Reply:
`self.fields` cannot be set. It's a getter property.